service/auth.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2022 Fraunhofer AISEC
```

Comment:
why is this package not in the corresponding auth subfolder?

Reply:
Because this is not part of the auth *service*, but rather functions related to the authentication between services and/or clients. I was also thinking moving it into a `service/common` or `service/middleware` package. But it is definitely NOT a part of `service/auth`, which contains our authentication service (or we need to rename that package).

Reply:
Thx for explanation. Then, I would advocate the `service/common` approach. Or maybe even `pkg` like stated in [this stackoverflow answer](https://stackoverflow.com/a/48872252/16445647).

OT: Looking at the 2nd alternative of this answer, I start thinking about putting the protobuf files (from `proto` folder) and generated swagger files (`openapi` folder) into the `api` directory as well for a more compact view of the project.

Reply:
> Thx for explanation. Then, I would advocate the `service/common` approach. Or maybe even `pkg` like stated in [this stackoverflow answer](https://stackoverflow.com/a/48872252/16445647).
> 

I am against `pkg` for various reasons that we can discuss while drinking several beers. `service/common` or `service/middleware` is fine for me.

> OT: Looking at the 2nd alternative of this answer, I start thinking about putting the protobuf files (from `proto` folder) and generated swagger files (`openapi` folder) into the `api` directory as well for a more compact view of the project.

Yes. This is something that I wanted to do for a long time anyway. I will create a separate issue for that I think.